### ISSUE TITLE
Unable to register vhost to server

### ISSUE THREAD

**Issue by i-m-sac** [patrickpissurno/fastify-vhost#1]
```
  server.register(vhost, {
    upstream: 'http://localhost:10002',
    host: 'staging.xyz.com'
  })
```

So this is as simple as this but shows an error saying url expected type string 

```
internal/validators.js:107
    throw new ERR_INVALID_ARG_TYPE(name, 'string', value);
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:107:11)
    at Url.parse (url.js:155:3)
    at Object.urlParse [as parse] (url.js:150:13)
    at Object.common.setupOutgoing (..\node_modules\fastify-vhost\node_modules\http-proxy\lib\http-proxy\common.js:87:12)
```

In the file common.js in  http-proxy inside nodeModules of vhost, this is where the error occurs
 ```
var outgoingPath = !options.toProxy
    ? (url.parse(req.url).path || '')
    : req.url;
```
There is not key 'url', its inside req.raw

**Comment by patrickpissurno** [comment]
Hello! Thanks for reporting the issue.

I could confirm the bug and I'm working in a fix. It'll be released tonight (GMT-3). The currently published version on NPM works only with fastify@1.x.x.

Reason:
With fastify@2.x.x, there were some breaking changes in the 'onRequest' hook. In 1.x.x the function arguments were (req, res, next) whereas in 2.x.x they are (request, reply, next).

Sorry for the inconvenience!

**Comment by patrickpissurno** [comment]
The issue is fixed by the latest release (1.1.3).

You can run `npm i fastify-vhost@latest` in your repo to update it.

I'll close this issue. Feel free to reopen it in case it still doesn't work for you. Thanks!
